scripts/gdal/gdal_helper.py

```diff
@@ -36,6 +36,26 @@ def command_to_string(command: List[str]) -> str:
     return " ".join(command)
 
 
+def get_gdal_version() -> str:
+    """Return the GDAL version assuming all GDAL commands are in the same version of gdalinfo.
+
+    Raises:
+        GDALExecutionException: If the GDAL command fails.
+
+    Returns:
+        str: The GDAL version returned by GDAL.
+    """
+    gdal_env = os.environ.copy()
+    gdalinfo_version = ["gdalinfo", "--version"]
+    try:
+        proc = subprocess.run(gdalinfo_version, env=gdal_env, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+    except subprocess.CalledProcessError as cpe:
+        get_log().error("get_gdal_version_failed", command=command_to_string(gdalinfo_version), error=str(cpe.stderr, "utf-8"))
+        raise GDALExecutionException(f"GDAL {str(cpe.stderr, 'utf-8')}") from cpe
+
+    return proc.stdout.decode().strip()
```

Comment:
this is somewhat weird in a blocked scope lang this would fail as `proc` is not defined in this scope as it was only defined in the  `try` scope.

It would be best to put this into the `try`: scope

Reply:
Oh that's a mistake